platform/mds/mds-web/src/main/resources/webapp/js/directives.js

```diff
@@ -2570,6 +2581,134 @@
+            'boolean',
+            'break',
+            'byte',
+            'case',
+            'catch',
+            'char',
+            'class',
+            'const*',
+            'continue',
+            'default',
+            'do',
+            'double',
+            'else',
+            'enum',
+            'extends',
+            'false',
+            'final',
+            'finally',
+            'float',
+            'for',
+            'goto*',
+            'if',
+            'int',
+            'interface',
+            'instanceof',
+            'implements',
+            'import',
+            'long',
+            'native',
+            'new',
+            'null',
+            'package',
+            'private',
+            'protected',
+            'public',
+            'return',
+            'short',
+            'static',
+            'strictfp',
+            'super',
+            'synchronized',
+            'switch',
+            'synchronized',
+            'this',
+            'throw',
+            'throws',
+            'transient',
+            'true',
+            'try',
+            'void',
+            'volatile',
+            'while'
+        ];
+        return {
+            require: 'ngModel',
+            link: function(scope, element, attrs, ctrl) {
+                var validateReservedWords,
+                    result = false,
+                    checkReservedWords = function (inputValue) {
+                        result = false;
+                        $.each(RESERVED_WORDS, function (index, reservedWord) {
+                            if (reservedWord === inputValue) {
+                                result = true;
+                            } else {
+                                result = false;
+                            }
+                            return (!result);
+                        });
+                        return result;
+                    };
+
+                validateReservedWords = function (viewValue) {
+                    if (ctrl.$viewValue === '' || attrs.reservedWordsValidity === 'true' || !checkReservedWords(ctrl.$viewValue)) {
+                        // it is valid
+                        ctrl.$setValidity('reservedwords', true);
+                        return viewValue;
+                    } else {
+                        // it is invalid, return undefined (no model update)
+                        ctrl.$setValidity('reservedwords', false);
+                        return '';
+                    }
+                };
+
+                ctrl.$parsers.unshift(validateReservedWords);
+
+                scope.$watch("field.settings[1].value", function(newValue, oldValue) {
+                    if (newValue !== oldValue) {
+                        ctrl.$setViewValue(ctrl.$viewValue);
+                    }
+                });
+            }
+        };
+    });
+
+    directives.directive('colonValidity', function() {
+        var COLON_REGEXP = /[:]|[;]/;
```

Comment:
what are we validating here, just colons? What's the point, there are other illegal symbols. We should just check if the input is alphanumeric and not a reserved keyword.